rewriter/rewriter.cc

```diff
@@ -197,6 +198,17 @@ class Rewriterer {
         }
     }
 
+    // NOTE: this case differs from the `Assign` typecase branch in `postTransformClassDef` above,
+    // as it will apply to all assigns, not just those that are present in the RHS of a `ClassDef`.
```

Comment:
Do we actually want it to apply to all assigns?  It seems like doing this for assigns inside e.g. a block or a method definition is pointless and could lead to weird error messages (maybe?).

(I can believe that this is the way the rewriter previously worked, but I'm curious if we'd like it to work differently this time around.)

Reply:
Inside a block at the class level, it should be fine, because class definitions are already allowed there.

Inside a method is impossible for this rewriter to fire, because our parser treats constant assignments in method bodies as parse errors and recovers from them by changing the assign to a variable:

[→ View on sorbet.run](https://sorbet.run/?arg=--print=desugar-tree#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Adef%20example%0A%20%20X%20%3D%201%0Aend)

```ruby
# typed: true
extend T::Sig

def example
  X = 1
end
```

#### Observed output

```
class <emptyTree><<C <root>>> < (::<todo sym>)
  <self>.extend(<emptyTree>::<C T>::<C Sig>)

  def example<<todo method>>(&<blk>)
    <dynamic-const-assign> = 1
  end
end
editor.rb:5: dynamic constant assignment https://srb.help/2001
     5 |  X = 1
          ^
Errors: 1
```

> (I can believe that this is the way the rewriter previously worked, but I'm curious if we'd like it to work differently this time around.)

In fact it _didn't_ work this way last time, and changing it to work this way now makes it correctly detect cases that it couldn't before.

Overall I'm not super worried, because the protobuf-generated files have such a predictable shape.